sre-recipes/sandboxctl

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This CLI is the interface for sre recipes. It deploys and rolls back
+broken services as well as verifies that the user found the correct
+cause of the broken service.
+
+For information on how to run the CLI, run the following:
+`python3 sandboxctl --help`
+"""
+
+import logging
+import os
+from os.path import dirname, abspath
+from pkgutil import iter_modules
+from pathlib import Path
+from importlib import import_module
+from inspect import isclass
+import click
+from recipe import Recipe
+
+def get_valid_recipes(attributes):
+    """
+    Gets all valid recipes from the given list of tuples of attribute names
+    and attributes, and returns a map containing recipe names and recipe objects
+    """
+
+    recipe_objs = {}
+    recipe_num = 1
+    for attribute_name, attribute in attributes:
+        if isclass(attribute) and attribute is not Recipe and issubclass(attribute, Recipe):
+            globals()[attribute_name] = attribute
```

Comment:
This could use a comment. I'm not exactly sure if I know what's going on here

Reply:
I realized I actually don't need that line of code. I think if you want to reference the class name, you need to set the global variable, but since I'm storing an instance of each recipe in a map, I don't need to set it globally.